### ISSUE TITLE
Create CDash tests for Albany Catalyst features

### ISSUE THREAD

**Issue by gahansen** [SNLComputation/Albany#158]
Catalyst is not currently tested or is it's state tracked on the dashboards.

CC: @ibaned @ikalash @jewatkins

**Comment by jewatkins** [comment]
Let me try running the [LCM Catalyst](https://github.com/gahansen/Albany/tree/master/tests/small/LCM/Catalyst) test before I made commit https://github.com/gahansen/Albany/commit/33aeef00a5670482f6d002fae03b84479d954c76 because I'm sure that commit would have broken it.

**Comment by ikalash** [comment]
Per our discussion at today's Albany meeting, here is a plan forward:

1. @ibaned will get Catalyst compiling in Albany (prior to @jewatkins 's commit changing the data structures), and create a wiki pages documenting what versions of paraview / catalyst are required.
2. @jewatkins will get Catalyst compiling in up-to-date Albany by switching it to new data structures.  
3. @gahansen  will write / debug a Catalyst test.

After 1. and 2. are done, we will need to create tests for Catalyst - can discuss at a later point where to do this / who will do it.

**Comment by ikalash** [comment]
@gahansen : what is the status of catalyst?  We are talking about it on the Albany concall.  I don't think it's built in the nightlies, and I don't think we ever got around to writing tests.  As with the moertel stuff, @bartgol and I are wary keeping it around when it is not tested.

**Comment by gahansen** [comment]
@ibaned and @jewatkins - what is the status of the work listed above (@ikalash 7/31/2017 comment)?
If this is in my court - I'll go ahead and write a test, get it running, and commit it. If we are shy on either 1. or 2.; I would suggest that we tag catalyst and remove it for now until we have a better support model for it. Any thoughts?

**Comment by bartgol** [comment]
Discretizations are quite complex in Albany. To test them fully, we would need a few tests. PUMI has only 3 tests (at least without activating any physics package), and I already think they're too few to be confident the classes are not broken. If you have time to write 4-5 tests, then ok, otherwise we should relegate to a tag.

To be clear, the different tests should reach (besides the always used stuff like getMapT()), less common paths of the code. E.g., there should be a test with distributed parameters, a test with restart (if it is theoretically supported), a test with vector solution (possibly interleaved, if supported), some responses, mesh adaptation (if supported), refinement (if supported), or rebalancing (if supported).

I know, it sounds like a lot. But that's the bane of the discretization classes: they are huge, and have lots of responsibilities, which are not reached by all tests.

**Comment by jewatkins** [comment]
I think Dan took the lead on getting things to compile even after I merged the new data structures. This is the last commits related to Catalyst: https://github.com/SNLComputation/Albany/commit/37b91b6a843b20112efd5a9271952eb1c766c7d8 , https://github.com/SNLComputation/Albany/commit/9fbc457a4f2067e54604b25682b4cd2b36d1f29d

If we didn't create nightly builds or tests, I assume it's probably broken again. I'd be okay with a tag.

**Comment by ibaned** [comment]
Yep @jewatkins is right, I got it compiling at some point, but nothing I know of checks nightly that it still compiles. Also, I haven't run it nor do I know of someone running it, so it might not run correctly even if it compiles.

**Comment by gahansen** [comment]
I'll take a look at the health of this over the next couple of days and determine if it is possible to make a test as things stand.

**Comment by gahansen** [comment]
I took a shot at re-invigorating Catalyst support. There have been some changes to both Albany and the Catalyst capabilities in Paraview since we added this capability; it is pretty clear that it is not being used by anyone given the state of the build. It might also be challenging to add tests of this capability such that it does not get broken under future development.

Lets go ahead and tag this as things stand and revisit later as needs develop.
